### ISSUE TITLE
Valenoroi stops "painting" the map

### ISSUE THREAD

**Issue by seppeel** [ccoors/Valeronoi#11]
I own a Roborock S5 Max rooted and running with Valetudo 2021.03.0.
When i start Cleaning and Valenoroi Recording -> Everything looks fine and the Map is starting to get "painted".
But after something between 1500 and 2000 Measurements the map stops getting painted. I see the "Measurement" and "Unique Places" Counters increasing. The "Lastest Signal" is changing to the (i would say) correct value.
Is there something like an upper limit for Measurements Valenoroi is able to "paint"?
I did this two times, it was always the same result.

![grafik](https://user-images.githubusercontent.com/31655676/113116506-cdce1f80-920d-11eb-97f8-bd18fda08083.png)

**Comment by ccoors** [comment]
That´s weird. I want to take a look at the raw data. Can you save and attach the file from Valeronoi? Or send it by Mail/Telegram, if you´re not comfortable sharing it on GitHub.

**Comment by ccoors** [comment]
When I load that file in Valeronoi it looks like this (which I´d say looks fine):

![image](https://user-images.githubusercontent.com/8963459/113118004-50a3aa00-920f-11eb-8daa-8e459c4d9c44.png)

Did you enable OpenGL?

**Comment by ccoors** [comment]
Indeed. Sorry for the inconvenience, I can´t really reproduce it myself because my apartment is quite small :) I´ll have to wait until Valetudo supports virtual maps in MockRobot.

**Comment by seppeel** [comment]
let me know if i can provide anything to help.
thanks for your work.

**Comment by ccoors** [comment]
I just released Valeronoi 0.1.3 with a new logging window (Help -> Log). If you get the chance, could you try it again?

Does the log continue to print "Requesting generation of Voronoi segments" and "Calculated new Voronoi segments" when the painting stops?

This will help narrow the issue down. The Voronoi diagram calculation is done in a separate thread because it's quite expensive. I suspect the thread just hangs after some time. If no more "Calculated new Voronoi segments" appear, that's the case.

**Comment by seppeel** [comment]
I will deliver some information as soon as our house looks like it needs some robo cleaning ;-)

**Comment by albpara** [comment]
Has similar issues with two different maps today, haven't saved anything that could provide help apart form the attached image. I will try to reproduce the problem again:

![1stfloor](https://user-images.githubusercontent.com/2099670/116929717-1c346b00-ac5f-11eb-9b20-5b500bc6cc7a.png)

**Comment by seppeel** [comment]
I was not able to reproduce it after updating to 0.1.3 with the last cleaning.

**Comment by winebob** [comment]
Having a similar issue here. I watched for a while live when it was near the dock and it was painting correctly. Came back at the end and there was almost no color showing. Saved, reloaded, and it looks like this:
![image](https://user-images.githubusercontent.com/16059714/118276374-3525f300-b48d-11eb-8b4a-2003f0f8df91.png)

[Upper Floor.zip](https://github.com/ccoors/Valeronoi/files/6478994/Upper.Floor.zip)

**Comment by ccoors** [comment]
Sorry for taking so long to look into this. I have actually found some issues in the code, which have been addressed in #46. The fixes will be included in the next release.
